galley/pkg/config/source/kube/inmemory/kubesource.go

```diff
@@ -196,24 +199,65 @@ func (s *KubeSource) RemoveContent(name string) {
 func (s *KubeSource) parseContent(r schema.KubeResources, name, yamlText string) ([]kubeResource, error) {
 	var resources []kubeResource
 	var errs error
-	for i, chunk := range kubeyaml.Split([]byte(yamlText)) {
-		chunk = bytes.TrimSpace(chunk)
+
+	reader := strings.NewReader(yamlText)
+	readCloser := ioutil.NopCloser(reader)
+	decoder := yaml.NewDocumentDecoder(readCloser)
+
+	buf := make([]byte, 4096)
```

Comment:
Where does this number come from?

Reply:
my brain!

it's an almost-certainly-outdated rule-of-thumb number for IO reads that I have no data to back up. What would you suggest?